src/core/PlaneWaveBasis.jl

```diff
@@ -0,0 +1,176 @@
+using FFTW
+using StaticArrays
+
+struct PlaneWaveBasis{T <: Real}
+    # Lattice and reciprocal lattice vectors in columns
+    lattice::SMatrix{3, 3, T, 9}
+    recip_lattice::SMatrix{3, 3, T, 9}
+    unit_cell_volume::T
+    recip_cell_volume::T
+
+    # Selected energy cutoff at construction time
+    Ecut::T
+
+    # Size of the rectangular Fourier grid used as the density basis
+    # and the k-point-specific (spherical) wave function basis
+    # The wave vectors are given in integer coordinates.
+    grid_size::SVector{3, Int}
+    idx_DC::Int  # Index of the DC component
+    kpoints::Vector{SVector{3, T}}
+    wfctn_basis::Vector{Vector{SVector{3, Int}}}
```

Comment:
Need to explain here when you have multi-d arrays: is it `wfctn_basis[ik][iG]` or `wfctn_basis[iG][ik]`? Maybe use `wf` for wave functions (otherwise I'll never remember what particular letters I should put here)

Reply:
Renamed to `wf_basis` and added clearifying comments.